nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -162,6 +176,60 @@ async fn run() -> Result<()> {
                 continue;
             }
 
+            // Check assigner in whitelist
+            if !assigner_whitelist.contains(&assign_event.sender) {
+                tracing::error!("assigner not in whitelist");
+                continue;
+            }
+
+            if let Ok(Some(assignee)) = get_single_tag_entry('p', &assign_event.tags) {
+                // Check assignee is Result sender
+                if let Ok(assignee_hex) = hex::decode(assignee) {
+                    if let Some(assignee_account) = Sender::from_bytes(&assignee_hex) {
+                        if assignee_account != sender {
+                            tracing::error!("assignee mismatch");
+                            continue;
+                        }
+
+                        // Verify assign_event signature
+                        let expected_hash = assign_event.to_id_hash();
+                        if expected_hash != assign_event.id {
+                            tracing::error!("invalid assign_event id");
+                            continue;
+                        }
+                        if let Err(_) = assign_event
+                            .sender
+                            .validate_signature(expected_hash, &assign_event.sig)
+                        {
+                            tracing::error!("invalid assign_event sig");
+                            continue;
+                        }
+                    } else {
+                        tracing::error!("invalid p tag in assign_event");
+                        continue;
+                    }
+                } else {
+                    tracing::error!("invalid p tag in assign_event");
+                    continue;
+                }
+            } else {
+                tracing::error!("missing p tag in assign_event");
+                continue;
+            }
+
+            // Check repeated assign event id
+            let exist = collection
```

Comment:
Not sure why we check the DB. We should just check the Result is valid and if the Result is already in our DB, write will fail. We pay only workers whose Result is successfully written on DB.

Reply:
I think there is no duplicate validation rules in DB?